lib/sparql_to_sw_solr/instance_solr_doc.rb

```diff
@@ -75,6 +75,13 @@ def solution_values_for_binding(solutions, binding_symbol)
       end
     end
 
+    def concatenate_values(val1, separator, val2)
+      result = "#{val1}" \
+      "#{present?(val1) && present?(val2) ? separator : nil}" \
+      "#{val2}"
+      result unless result.empty?
+    end
```

Comment:
This and the specs are great, but I think this is all taken care of by the ruby `String#join` method already.  It's not required, but if your curious, please try to replace the body of the method with something like the following and see if it still passes all the specs:
```ruby
[val1, val2].join(separator)
```
(I'm not sure what happens when the `separator` is `nil`.)